hibernate-core/src/main/java/org/hibernate/tuple/entity/AbstractEntityTuplizer.java

```diff
@@ -569,6 +565,20 @@ protected boolean shouldGetAllProperties(Object entity) {
 			// if the attribute is not lazy (bytecode sense), we can just use the value from the instance
 			// if the attribute is lazy but has been initialized we can just use the value from the instance
 			// todo : there should be a third case here when we merge transient instances
+			if(enhancedForLazyLoading) {
+				final BytecodeLazyAttributeInterceptor interceptor = enhancementMetadata.extractLazyInterceptor(entity);
+				if (interceptor == null && lazyAttributesMetadata.isLazyAttribute( propertyName ) && getters[j].get(entity) == null) {
+					result[j] = LazyPropertyInitializer.UNFETCHED_PROPERTY;
+					continue;
+				}
+			}
+			else {
+				Type type = entityMetamodel.getPropertyTypes()[j];
+				if(type.isCollectionType() && ((CollectionType)type).getReturnedClass().isAssignableFrom(Serializable.class) && !((CollectionType)type).hasHolder() && entityMetamodel.isLazy() && getters[j].get(entity) == null) {
```

Comment:
Copying the [conversation from the original PR](https://github.com/hibernate/hibernate-orm/pull/4339#discussion_r742617957).

From @yrodiere:

> Checking that the collection type is serializable? This looks like an awfully specific condition. Can you explain why you're restricting this code to just serializable collections?

From @busitech:

> Hibernate still supports legacy use cases.  LazyPropertyInitializer.UNFETCHED_PROPERTY gets assigned to all collection properties while they are uninitialized, and Serializable is the only interface it implements.  If this marker value cannot be assigned to a collection property, then we can't proceed with marking it unfetched.

From @busitech:

> Do you know of a better test for detecting when any of these collection types are in use, which can not be used with LazyPropertyInitializer?

Reply:
> these collection types are in use

What I was trying to ask was, what are "these" collection types?

Is this code to handle collections mapped to a single column as a serialized blob?

Reply:
No, this filter is the opposite of a very specific filter. It is a very broad, inclusive test, only omitting a very select few collection types like ArrayList on legacy test cases. This does not filter a few things in, it kicks a few things out that fail legacy test cases with IllegalArgumentException. This filter lets almost every other type of collection be handled here.

The only ones that get omitted are those to which LazyPropertyInitializer.UNFETCHED_PROPERTY cannot be assigned (and because it implements Serializable as it's only interface, so there's no other good way to test compatibility).

AbstractPersistentCollection implements Serializable, which is the superclass for almost every persistent collection type in ORM, but just to be sure not to break any other test cases, I simply used the interface for LazyPropertyInitializer.UNFETCHED_PROPERTY as the test.